crates/rome_js_formatter/src/js/expressions/call_arguments.rs

```diff
@@ -14,24 +21,239 @@ impl FormatNodeFields<JsCallArguments> for FormatNodeRule<JsCallArguments> {
+                                        )
+                                    }))
+                                    .entries(iter)
+                                    .finish()
+                            }),
+                            r_leading_trivia,
+                            r_paren,
+                            r_trailing_trivia
+                        ]
+                    )
+                } else {
+                    write!(
+                        f,
+                        [
+                            l_leading_trivia,
+                            l_paren,
+                            l_trailing_trivia,
+                            format_with(|f| {
+                                // special formatting of the last element
+                                let mut iter = separated.iter();
+                                // SAFETY: check on the existence of at least one argument are done before
+                                let last = iter.next_back().unwrap();
+
+                                f.join_with(&space_token())
+                                    .entries(iter)
+                                    .entry(&format_with(|f| {
+                                        write!(
+                                            f,
+                                            [group_elements(&format_args![last, expand_parent()])]
+                                        )
+                                    }))
+                                    .finish()
+                            }),
+                            r_leading_trivia,
+                            r_paren,
+                            r_trailing_trivia
+                        ]
+                    )
+                }
+            });
+
+            // This is the version of where all the arguments are broken out
+            let all_arguments_expanded = format_with(|f| {
+                // this formatting structure replicates what we have inside the `format_delimited`
+                // function, but here we use a different way to print the trailing separator
+                write!(
+                    f,
+                    [
+                        &l_leading_trivia,
+                        &l_paren,
+                        &group_elements(&format_args![format_with(|f| {
+                            write!(
+                                f,
+                                [
+                                    &soft_block_indent(&format_args![
+                                        &l_trailing_trivia,
+                                        format_with(|f| {
+                                            fmt_arguments_multi_line(
+                                                separated.iter(),
+                                                separated.len(),
+                                                f,
+                                            )
+                                        }),
```

Comment:
Nit: It may be slighlty nicer to use if `format_arguments_multi_line` returns an object that implements `Format` rather than doing the formatting right away
```suggestion

                                            format_arguments_multi_line(
                                                separated.iter(),
                                                separated.len(),
                                            ),
```